### PR TITLE
Dockerise application

### PR DESCRIPTION
**Pull request by saliceti**
### Context
Suggest deployment best practices

### Changes proposed in this pull request
Add Dockerfile, docker-compose, Makefile

### Guidance to review
Run the new commands in Readme

### CODE REVIEW

**Comment by saliceti** [comment]
[Dockerise rails-boilerplate](https://trello.com/c/rotrr26H/201-dockerise-rails-boilerplate)

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+# Add postgres driver library
+# Add the timezone as it's not configured by default in Alpine
+RUN apk add --update --no-cache libpq tzdata && \
+  cp /usr/share/zoneinfo/Europe/London /etc/localtime && \
+  echo "Europe/London" > /etc/timezone
+
+# Copy files generated in the builder image
+COPY --from=builder /app /app
+COPY --from=builder /usr/local/bundle/ /usr/local/bundle/
```
Comment (vigneshmsft) [inline Dockerfile:60]: Add default CMD as well?

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,12 @@
+.PHONY: build-local-image
+build-local-image:
+	docker pull dfedigital/govuk-rails-boilerplate:builder-latest || true
```
Comment (vigneshmsft) [inline Makefile:3]: can use buildx?

File: docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,30 @@
+version: '3.4'
```
Comment (vigneshmsft) [inline docker-compose.yml:1]:
```suggestion
version: '3.6'
```

Comment (saliceti) [reply docker-compose.yml]: Done

File: Makefile (hunk 2 above)
Comment (saliceti) [reply Makefile]: Done

File: Dockerfile (hunk 1 above)
Comment (saliceti) [reply Dockerfile:60]: There are many variations for running the image:
- webapp or sidekiq
- with db:setup or db:migrate
- with a database or not

I thought of not adding CMD on purpose so it is set explicitly when it is used, for instance in the docker-compose file

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+# build-base: complication tools for bundle
+# yarn: node package manager
+# postgresql-dev: postgres driver and libraries
+RUN apk add --no-cache build-base yarn postgresql-dev
+
+# Install bundler to run bundle exec
+# This should be the same version as the Gemfile.lock
+RUN gem install bundler:2.1.4 --no-document
+RUN bundle config set without 'development test'
```
Comment (vigneshmsft) [inline Dockerfile:20]: `without test` ?
will gems required for running tests be installed?

**Review by vigneshmsft — APPROVED** [review]

File: Dockerfile (hunk 4 above)
Comment (saliceti) [reply Dockerfile:20]: This image will be used to deploy to paas, no need for test gems. If they are required later on we can create a story to create a separate image

**Pull request merged**